File: frontend/types/deliverable.ts
export type DeliverableStatus =
  | "not_started"
  | "draft_ready"
  | "review_ready"
  | "approval_pending"
  | "approved"
  | "delivered"
  | "superseded";

export type DeliverableType =
  | "social_post"
  | "banner"
  | "presentation"
  | "media_kit"
  | "brand_asset"
  | "institutional_piece"
  | "thumbnail"
  | "campaign_kit"
  | "editorial_piece";

export interface Deliverable {
  id: string;
  design_request_id: string;
  deliverable_type: DeliverableType;
  name: string;
  format: string;
  channel?: string | null;
  status: DeliverableStatus;
  delivery_channel?: string | null;
  final_version_id?: string | null;
  approved_at?: string | null;
  delivered_at?: string | null;
  created_at: string;
  updated_at: string;
}

export interface DeliverableListResponse {
  data: Deliverable[];
  meta: { count: number };
  error: null;
}

File: backend/app/schemas/deliverable.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import DeliverableStatus, DeliverableType
from app.schemas.common import TimestampedSchema


class DeliverableBase(BaseModel):
    design_request_id: UUID
    deliverable_type: DeliverableType
    name: str
    format: str
    channel: str | None = None
    status: DeliverableStatus
    delivery_channel: str | None = None
    final_version_id: UUID | None = None


class DeliverableCreate(DeliverableBase):
    pass


class DeliverableUpdate(BaseModel):
    status: DeliverableStatus | None = None
    delivery_channel: str | None = None
    final_version_id: UUID | None = None


class DeliverableRead(DeliverableBase, TimestampedSchema):
    id: UUID
    approved_at: datetime | None = None
    delivered_at: datetime | None = None


class DeliverableListResponse(BaseModel):
    data: list[DeliverableRead]
    meta: dict[str, int]
    error: None = None
